include/mbedtls/check_config.h

```diff
@@ -359,7 +359,8 @@
 
 #if defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED) &&                    \
     ( !defined(MBEDTLS_PK_HAVE_JPAKE) ||                                    \
-      !defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED) )
+      !(defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED) || \
```

Comment:
I'm afraid it's a bit more complex: TLS will only take advantage of the presence of the curve in PSA is `USE_PSA` is defined. So, I think this should be
```c
!((!defined(MBEDTLS_USE_PSA_CRYPTO) && defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED)) || \
  (defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED) && defined(PSA_WANT_ECC_SECP_R1_256)))
```
Obviously, if we find ourselved using this expression too often (and perhaps once is already too often for readability?) we could define a macro `MBEDTLS_PK_HAVE_CURVE_xxx` like we did for ECDH and the others.